### DISCUSSION TITLE
tar_map_rep() over tibble with vector of lists

### DISCUSSION THREAD

**Discussion by mitchell-thomann** [ropensci/tarchetypes#127] [Help]
I am trying to use tar_map_rep() over values in tibble with a vector that is a list of parameters.  See below for an example of a _targets.R file:

```r
scenario_tibble <- tibble(
  scenario = list(all_viable = all_viable, two_viable = two_viable),
  scenario_label = c("all_viable", "two_viable")
)

list(
  #run sim 
  tar_map_rep(
    sims,
    trialsim_twoarms_bayes(
      pars_list = scenario,
      label = scenario_label
    ),
    batches = batches,
    reps = reps,
    values = scenario_tibble
  )
)
```

The error received is:
  no applicable method for 'isSymmetric' applied to an object of class "NULL"

This indicates that the parameters in the lists are not being passed to the function correctly as there is an additional level to the list than what is expected.  Before reworking the function - is there an error in the way that I am passing such lists in tar_map_rep()?

**Comment by wlandau** [comment]
`tar_manifest()` can show you the commands that `tar_map_rep()` generates, including the specific calls to `trialsim_twoarms_bayes()`.

```r
# _targets.R file
library(targets)
library(tarchetypes)

scenario_tibble <- tibble::tibble(
  scenario = list(all_viable = c(1, 2), two_viable = c(3, 4)),
  scenario_label = c("all_viable", "two_viable")
)

list(
  tar_map_rep(
    sims,
    trialsim_twoarms_bayes(
      pars_list = scenario,
      label = scenario_label
    ),
    batches = 1,
    reps = 1,
    values = scenario_tibble
  )
)
```

```r
R console
writeLines(tar_manifest()$command[[2]])
#> tarchetypes::tar_rep_run(command = tarchetypes::tar_append_static_values(object = trialsim_twoarms_bayes(pars_list = c(1, 
#>      2), label = "all_viable"), values = list(scenario = c(1, 
#>      2), scenario_label = "all_viable")), batch = sims_batch, 
#>      reps = 1, iteration = "vector")
```

Looking at the output above, you can try `trialsim_twoarms_bayes(pars_list = c(1, 2), label = "all_viable")` and adjust the elements of `scenario_tibble`.

Hopefully that works. If not, it would help if you would post a [reproducible example](https://reprex.tidyverse.org) I can run, like in https://github.com/ropensci/tarchetypes/discussions/126. Without `trialsim_twoarms_bayes()` etc., it's hard to say what would cause that `isSymmetric` error.

**Comment by mitchell-thomann** [comment]
Thank you - I identified the origin of the error message with this help.  When mapping over the list elements, the values are maintained but the element names and class types were not.  I fixed this by reassigning these within the function itself but any input on better practice for this would be appreciated.

**Reply by wlandau** [reply]
Glad you figured it out. Looks like your situation was just like https://github.com/ropensci/tarchetypes/discussions/105.

`tar_map()` and `tar_map_rep()` create expressions for whole collections of targets. In other words, they create R code by substituting objects from the `values` argument (in you case, `scenarios_tibble`). It is not always trivial to substitute an element of `scenarios_tibble` into a valid R expression and get a valid R expression back. That's why it's useful to look at the `tar_manifest()` call and verify that the correct R commands were generated.
